client/src/components/GetDetails/index.js

```diff
@@ -129,7 +130,22 @@ class GetDetails extends Component {
         pattern: this.state.selected_patterns,
       };
 
-      if (cookie) {
+      if (inputs.typeId === '') {
+        axios.post('/add-type', { name: inputs.type.name, shortcut: 'New Type' }).then((res) => {
+          const { typeId } = res.data;
+          this.setState({ typeId }, () => {
+            if (cookie) {
+              axios.post('/add-item', inputs).then(({ data }) => {
```

Comment:
if you did not get what I mean, check this endpoint on BE, match FE with BE please

Reply:
Yeah will get around to do it in the next pull request, don't worry!